Enhancement/improve drtsafe error

mnlevy1981 opened the pull request:
Add DIC, PT, TA, and SIT to error message when `drtsafe()` does not converge. I did this by creating a new local variable of type `co2calc_state_type` in `marbl_co2calc_surf()` and `marbl_comp_CO3terms()` (copies of `dic_in`, `pt_in`, `ta_in`, and `sit_in`) to ensure that the log contains the values passed by the GCM rather than the value after MARBL limits the range by applying a minimum value check.

I don't really like how this leads to passing both `marbl_co2calc_state_in` and `marbl_co2calc_state` as `intent(in)`, so maybe better variable names are called for?

If I force an error by setting the `del_ph` to a very small number then the error reads:

```
Message from (lon, lat) ( 323.300, -77.981), which is global (i,j) (1, 2)
(marbl_co2calc_mod:drtsafe) it = 4
(marbl_co2calc_mod:drtsafe) x1,f =  0.7588892E-008 0.3275306E-006
(marbl_co2calc_mod:drtsafe) x2,f =  0.7588948E-008 0.3262787E-006
MARBL ERROR (marbl_co2calc_mod:drtsafe): bounding bracket for pH solution not found
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) dic =  0.2230582E+004
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) ta =  0.2373427E+004
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) pt =  0.1780354E+001
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) sit =  0.5790140E+002
```

Addresses #151

mnlevy1981 committed:
Pass uncapped co2calc_state vals to drtsafe

If solver does not converge, we want to write these values to the log. Note
that this is a first pass at a simple introduction of the uncapped values; next
commit will try to be smarter in terms of passing these values through the
calling tree.

mnlevy1981 committed:
Change how uncapped co2calc_state is handled

instead of passing dic_in, ta_in, pt_in, and sit_in to comp_htotal(), we now
put all those variables in co2calc_state_in; this lets us write those values to
the error log in drtsafe even after co2calc_state is updated (min value
applied)

mnlevy1981 committed:
Need to actually add log_message to log

Wrote text to string, did not add string to marbl_status_log

klindsay28 reviewed (commented):
I'd like to see temp and salt values added.

mnlevy1981 committed:
Add T & S to drtsafe error message

Added temp and salt to co2calc_state_type so that they are available in
co2calc_state_in in drtsafe(); this also let me reduce the argument list
in comp_co2calc_coeffs() and comp_htotal()

mnlevy1981 commented:
New error message:

```
Message from (lon, lat) ( 323.300, -77.981), which is global (i,j) (1, 2)
(marbl_co2calc_mod:drtsafe) it = 4
(marbl_co2calc_mod:drtsafe) x1,f =  0.7588892E-008 0.3275306E-006
(marbl_co2calc_mod:drtsafe) x2,f =  0.7588948E-008 0.3262787E-006
MARBL ERROR (marbl_co2calc_mod:drtsafe): bounding bracket for pH solution not found
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) dic =  0.2230582E+004
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) ta =  0.2373427E+004
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) pt =  0.1780354E+001
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) sit =  0.5790140E+002
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) temp = -0.1228945E+001
MARBL ERROR (marbl_co2calc_mod:drtsafe): (marbl_co2calc_mod:drtsafe) salt =  0.3430182E+002
```

Note that having `temp` and `salt` in `co2calc_state_type` means we don't need to pass them to `comp_htotal` and `comp_co2calc_coeffs`, which is nice.

mnlevy1981 committed:
Forgot to init state%temp and %salt

Had initialized in marbl_co2calc_surf() but not marbl_comp_CO3terms()

klindsay28 reviewed (approved):
looks good
